.github/workflows/ci-linux.yml

```diff
@@ -77,13 +77,26 @@ jobs:
         run: |
           export cache_dir="${{ env.CACHE_DIR }}"
           export CCACHE_COMPILERCHECK="string:$(clang --version)"
-          bash build_tools/ci/build_test_cpp.sh
+          bash build_tools/build_llvm.sh
+          rm -rf llvm-build
+          export llvm_install_dir=$PWD/llvm-install
```

Comment:
> **EDIT**: in fact we actually need this for HSA (because ROCR requires an up-to-date Clang)

Do you need to build that from source yourself here, or could that come from official/upstream LLVM [nightly] builds? I guess https://llvm.org/builds/ doesn't include Windows (any longer), but https://apt.llvm.org/ exists for Linux (Debian/Ubuntu) at least.

Reply:
in ROCR i manged to get away with 

https://github.com/nod-ai/ROCR-Runtime/blob/amd-staging/.github/workflows/ci-linux.yml#L49-L54

```
      - name: Setup Cpp
        uses: aminya/setup-cpp@v1
        with:
          compiler: llvm-18
          cmake: true
          ninja: true
```

which jams a compatible ubuntu release of llvm into an almalinux runner. to make it work i had to install `sudo` on there. here, where we're building on nod-shared-builder-whatever, which is almalinux, it doesn't work because you get a complaint that "root isn't in the sudoers file" 🤷. I didn't try too hard because really ROCR is a cherry on-top and what i wanted here was to be able to reuse the LLVM builds that currently get thrown away.